pkg/lock/file.go

```diff
@@ -179,10 +184,12 @@ func NewLock(path string, lockType LockType) (*FileLock, error) {
 	var stat syscall.Stat_t
 	err = syscall.Fstat(lfd, &stat)
 	if err != nil {
+		l.Close()
 		return nil, err
 	}
 	// Check if the file is a regular file
 	if lockType == RegFile && !(stat.Mode&syscall.S_IFMT == syscall.S_IFREG) {
+		l.Close()
```

Comment:
As per comment, I'd only keep these two chunks here in `NewLock` and adjust the docstring accordingly.